presto-main/src/main/java/com/facebook/presto/execution/buffer/PagesSerde.java

```diff
@@ -39,56 +41,81 @@
     private static final double MINIMUM_COMPRESSION_RATIO = 0.8;
 
     private final BlockEncodingSerde blockEncodingSerde;
-    private final Optional<Compressor> compressor;
-    private final Optional<Decompressor> decompressor;
+    @Nullable private final Compressor compressor;
```

Comment:
Why is this change?

Reply:
To avoid unnecessary allocations and indirection. Since spilling can be invoked as a result of memory revocation and one `PagesSerde` is created per `SingleStreamSpiller`, I thought it better to use nullable references instead, but I can revert it back if you prefer. I have no data to support this change being measurable in reducing memory pressure or improving performance.

Reply:
@pettyjamesm I think these kind of optimizations make sense only when operation is happening for each row (e.g. number of operations is proportional to the amount of data being processed). It seems like this is not the case here. It is always hard to tell without a benchmark. If you feel like this is a necessary optimization, add a separate commit which applies it to compressor and decompressor before encryption commits.

Reply:
I'll back the change out and use `Optional`